src/cfn-resource-generator.ts

```diff
@@ -0,0 +1,144 @@
+import * as camelcase from 'camelcase';
+import * as j2j from 'json2jsii';
+
+export function sanitizeTypeName(typeName: string) {
+  const parts = typeName.split('::');
+  return parts.map(part => part.substr(0, 1).toUpperCase() + part.substr(1).toLowerCase()).join('');
+}
+
+export class CfnResourceGenerator {
+  sanitizedTypeName: string;
+
+  constructor(private readonly typeName: string, private readonly typeDef: AWS.CloudFormation.DescribeTypeOutput, private readonly schema: any) {
```

Comment:
Avoid leaking `DescribeTypeOutput` in here. Instead, just define an `Options` type with the relevant fields. In this case, seems like you just need the `sourceUrl` and `schema`.